texas-hearing-institute/pages/Onboarding1.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useEffect } from 'react'
+import { View,Text,Button, Pressable, SafeAreaView } from 'react-native'
+import tw from 'tailwind-react-native-classnames';
+import styles from './OnboardingStyle';
+import { SvgXml } from 'react-native-svg';
+import leftArrow from '../icons/leftarrow';
+import ProgressBar from '../components/ProgressBar/ProgressBar';
+import MarkdownText from '../components/MarkdownText/MarkdownText';
+import ToggleGridButtons from '../components/ToggleGridButtonsComponent/ToggleGridButtons';
+import {consonants,prompt1, storageKey} from './Constants'
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+const Onboarding1 = ({navigation}:any) => {
+
+    // hashmap/dictionary to keep track of all consonants and their toggled state, update via useState to rerender componetns
+    const [itemsSelected, setItemsSelected] = useState(Array(consonants.length).fill(false));
+
+    // Store selection
+    const storeItemSelection = async (storageKey: string, items: string[], itemsSelected: boolean[]) => {
+        let itemSelectionMap = new Map<string, boolean>();
+        for (let i = 0; i < items.length; i++) {
+            itemSelectionMap.set(items[i], itemsSelected[i]);
+        }
+        try {
+            await AsyncStorage.setItem(storageKey, JSON.stringify(Array.from(itemSelectionMap.entries())));
+        } catch (e) {
+            console.log(e);
+        }
+    }
+
+    // Retrieve selection
+    const retrieveItemSelections = async (storageKey: string, items: string[]): Promise<boolean[]> => {
+        try {
+            const itemSelectionJson = await AsyncStorage.getItem(storageKey);
+            let itemSelectionMap = new Map<string, boolean>();
+
+            if (itemSelectionJson !== null) {
+                const itemSelectionArray: [string, boolean][] = JSON.parse(itemSelectionJson);
+                itemSelectionMap = new Map(itemSelectionArray);
+            }
+
+            const itemsSelected: boolean[] = items.map(item => itemSelectionMap.get(item) ?? false);
+            return itemsSelected;
+
+        } catch (e) {
+            console.log(e);
+            return items.map(() => false);
+        }
+    }
+
+    // Load stored selection statuses when the component mounts
+    useEffect(() => {
+        const loadStoredSelections = async () => {
+            const storedSelections = await retrieveItemSelections(storageKey, consonants);
+            setItemsSelected(storedSelections);
+        };
+        loadStoredSelections();
+    }, []); // The empty array ensures this effect runs once on mount
+
+    // This effect triggers every time 'itemsSelected' changes, storing the new state
+    useEffect(() => {
+        storeItemSelection(storageKey, consonants, itemsSelected);
+    }, [itemsSelected]); // Re-run this effect when 'itemsSelected' changes
+
+
+
+    return (
+        <SafeAreaView style={{
```

Comment:
SafeAreaView is iOS specific - talk with Paul and Vincent for how they made it Android safe too with React Navigation